scripts/linux/nv/gitlab_test_valgrind.sh

```diff
@@ -0,0 +1,126 @@
+    "olp-cpp-sdk-dataservice-read-test"
+    "olp-cpp-sdk-dataservice-write-test")
+source ${FV_HOME}/olp-common.variables          # variables common for all tests
+if [ -z $production_service_secret ] || [ -z $production_service_id ] ; then
+    echo "FATAL FAILURE: production_service_id and production_service_secret are not defined. Have you defined env.variables on GitLab?"
+    exit 1
+fi
+ulimit -c unlimited # for core dump backtrace
+
+cd build
+
+# Start local server
+node ${REPO_HOME}/tests/utils/mock_server/server.js & export SERVER_PID=$!
+
+# Node can start server in 1 second, but not faster.
+# Add waiter for server to be started. No other way to solve that.
+# Curl returns code 1 - means server still down. Curl returns 0 when server is up
+RC=1
+while [[ ${RC} -ne 0 ]];
+do
+        set +e
+        curl -s http://localhost:3000
+        RC=$?
+        sleep 0.15
+        set -e
+done
+echo ">>> Local Server started for further functional test ... >>>"
+
+### Running two auto-test groups
+for test_group_name in integration functional
+do
+{
+    if [[ ${test_group_name} == "functional" ]] ; then
+        EXCEPTION="--gtest_filter=-ArcGisAuthenticationTest.SignInArcGis:FacebookAuthenticationTest.SignInFacebook:DataserviceReadVersionedLayerClientTest.Prefetch:DataserviceReadVersionedLayerClientTest.PrefetchWithCancellableFuture"
+    elif [[ ${test_group_name} == "integration" ]] ; then
+        EXCEPTION="--gtest_filter=-DataserviceReadVersionedLayerClientTest.PrefetchTilesBusy:DataserviceReadVersionedLayerClientTest.PrefetchTilesWithCancellableFuture:DataserviceReadVersionedLayerClientTest.PrefetchTilesWithCache:DataserviceReadVersionedLayerClientTest.Prefetch"
+    else
+        EXCEPTION=""
+        FLAKY_CHECK=""  #FLAKY_CHECK="--gtest_repeat=10"
+    fi
+
+    source ${FV_HOME}/olp-cpp-sdk-${test_group_name}-test.variables
+    test_command="$REPO_HOME/build/tests/${test_group_name}/olp-cpp-sdk-${test_group_name}-tests --gtest_output=xml:${REPO_HOME}/reports/olp-cpp-sdk-${test_group_name}-test-report.xml ${EXCEPTION} ${FLAKY_CHECK}"
+    valgrind_command="valgrind --leak-check=full --track-origins=yes --show-leak-kinds=all --xml=yes"
+    if [[ ! -z "$misc_folder_path" ]]; then
+        valgrind_result_file_path="${REPO_HOME}/${misc_folder_path}/${test_group_name}_memcheck.xml"
+        valgrind_command="${valgrind_command} --xml-file=${valgrind_result_file_path}"
+    fi
+    test_command="export GLIBCXX_FORCE_NEW=; ${valgrind_command} ${test_command}"
+
+    echo "-----> Calling \"${test_command}\" for ${test_group_name} : "
+    eval "${test_command}"
+    result=$?
+    echo "-----> Finished ${test_group_name} - Result=${result}"
+    if [[ ${result} -ne 0 ]]; then
+        exit ${result}
+    fi
+}
+done
+
+### Running all unit groups
+for test_name in "${TEST_TARGET_NAMES[@]}"
+do
+{
+    if [[ ${test_name} == "olp-cpp-sdk-core-test" ]] ; then
+        EXCEPTION="--gtest_filter="-DefaultCacheTest.BadPath""
+        else
```

Comment:
Iteration seems to not be correct for `else`

Reply:
Do you mean I need to remove 1 tab before _else_ ?

Reply:
So it looks to me.

Reply:
Done. Thanks.